### PR TITLE
mojang-minecraft/mojang-gametest: Minecraft 1.18.20 Beta API updates

### PR DESCRIPTION
**Pull request by mammerla**
Documentation: https://docs.microsoft.com/minecraft/creator/scriptapi/

New EntityComponents for tracking entity state
playSound
Vector Helpers

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Review by JakeShirley — APPROVED** [review]

File: types/mojang-gametest/index.d.ts [hunk 1]
```diff
@@ -841,7 +851,7 @@ export class SimulatedPlayer extends mojangminecraft.Player {
  * These well-known tags can be used to classify different
  * tests into suites to run.
  */
-// tslint:disable-next-line:no-unnecessary-class
+ // tslint:disable-next-line:no-unnecessary-class
```
Comment (jakebailey) [inline types/mojang-gametest/index.d.ts:854]: This change seems to have been made all over the place; was this intentional?

**Review by jakebailey — COMMENTED** [review]
It seems like much of this PR is reformatting the code to have a different indention and isn't related to the API update; I'm not sure if that was intentional or not on your part.

[The README](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes) says that things should be 4-space indented and provides a command if you'd like to restore it, though I do see that one file was already at a 2-space indention.

File: types/mojang-gametest/index.d.ts (hunk 1 above)
Comment (JakeShirley) [reply types/mojang-gametest/index.d.ts:854]: @jakebailey not intentional.  Let me double check our pipeline.  These files are autogenerated so I will check the spacing on our side.  Thanks for pointing this out!

**Comment by JakeShirley** [comment]
This is getting marked as "huge" (it is a lot of changes) but most of them are unifying the whitespace with the prettier spec listed in this repo.  Before we were unaware of this so hopefully this is the last large formatting change we'll need to push in.

**Review by jakebailey — APPROVED** [review]
At this point, I'm assuming all of the formatting changes are intentional this time. 😄

**Review by JakeShirley — APPROVED** [review]

**Comment by jakebailey** [comment]
Hah, race condition reply. Yeah, thanks for the update. Should make things a little better for future reviews.

**Pull request merged**